src/main/java/com/arpnetworking/metrics/mad/parsers/PrometheusToRecordParser.java

```diff
@@ -0,0 +1,245 @@
+                    );
+                    records.add(record);
+                }
+            }
+        } catch (final InvalidProtocolBufferException e) {
+            throw new ParsingException("Could not create Request message from data", data.getBody().toArray(), e);
+        } catch (final ConstraintsViolatedException | IllegalArgumentException e) {
+            throw new ParsingException("Could not build record", data.getBody().toArray(), e);
+        }
+        return records;
+    }
+
+    private ImmutableMap<String, ? extends Metric> createMetric(final String name, final Types.Sample sample, final Optional<Unit> unit) {
+        final Metric metric = ThreadLocalBuilder.build(
+                DefaultMetric.Builder.class,
+                p -> p
+                        .setType(MetricType.GAUGE)
+                        .setValues(ImmutableList.of(createQuantity(sample, unit)))
+                        .build()
+        );
+        return ImmutableMap.of(name, metric);
+    }
+
+    private Quantity createQuantity(final Types.Sample sample, final Optional<Unit> unit) {
+        return ThreadLocalBuilder.build(
+                Quantity.Builder.class,
+                p -> p
+                        .setValue(sample.getValue())
+                        .setUnit(unit.orElse(null))
+        );
+    }
+
+    private static String createUnitMapKey(final String name) {
+        return new StringBuilder(name).reverse().toString();
+    }
+
+    private final boolean _interpretUnits;
+
+    private static final ImmutableMap<String, Unit> UNIT_MAP = ImmutableMap.of(
+            createUnitMapKey("seconds"), Unit.SECOND,
+            createUnitMapKey("celcius"), Unit.CELCIUS,
+            createUnitMapKey("bytes"), Unit.BYTE,
+            createUnitMapKey("bits"), Unit.BIT
+    );
+    private static final ImmutableSet<String> PROMETHEUS_AGGREGATION_KEYS = ImmutableSet.of(
+            createUnitMapKey("total"),
+            createUnitMapKey("bucket"),
+            createUnitMapKey("sum"),
+            createUnitMapKey("avg"),
+            createUnitMapKey("count")
+    );
+
+    static class ParseResult{
+        private final Unit _unit;
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            final ParseResult that = (ParseResult) o;
+            return _unit == that._unit
+                    && _newName.equals(that._newName);
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                    .add("unit", _unit)
+                    .add("newName", _newName)
+                    .toString();
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(_unit, _newName);
+        }
+
+        private final String _newName;
```

Comment:
This field should be at the top with the other member field. 

Also, as suggested above it should probably just be renamed to `_name`.